packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m

```diff
@@ -422,9 +422,31 @@ - (NSDictionary *)constantsToExport
       [[PHImageManager defaultManager] requestAVAssetForVideo:asset
                                                       options:options
                                                 resultHandler:^(AVAsset * _Nullable asset, AVAudioMix * _Nullable audioMix, NSDictionary * _Nullable info) {
-        AVURLAsset *urlAsset = (AVURLAsset *)asset;
-        result[@"localUri"] = [[urlAsset URL] absoluteString];
-        resolve(result);
+        // Slow motion videos are returned as an AVComposition instance
+        if ([asset isKindOfClass:[AVComposition class]]) {
+            NSString *directory = [self.fileSystem.cachesDirectory stringByAppendingPathComponent:@"MediaLibrary"];
+            [self.fileSystem ensureDirExistsWithPath:directory];
+            NSString *videoOutoutFileName = [NSString stringWithFormat:@"slowMoVideo-%d.mov",arc4random() % 1000];
```

Comment:
```suggestion
            NSString *videoOutputFileName = [NSString stringWithFormat:@"slowMoVideo-%d.mov",arc4random() % 1000];
```

And is it possible to get the original file name?

Reply:
I don’t see any helpful method to get the “file name” yet:

PHAsset class: https://developer.apple.com/documentation/photokit/phasset